amti/actions/review.py

```diff
@@ -80,29 +82,52 @@ def review_hit(
                             assignment_id=assignment_id,
                             answers=answers_xml.toprettyxml()))
 
-                    approve = None
-                    while approve is None:
-                        user_input = input('Approve? [y/n]').strip().lower()
-                        if user_input in ['y', 'n']:
-                            approve = user_input == 'y'
+                    input_string = """Would you like to (a)ccept, (r)eject, (s)kip or
+                        (m)ark the assignment? [a/r/s/m]:"""
+                    while True:
+                        user_input = input(input_string).strip().lower()
+                        if user_input in ['a', 'r', 's']:
+                            break
+                        elif user_input == 'm':
+                            print('Assignment marked.')
+                            while True:
+                                user_input = input('Would you like to (a)ccept, (r)eject, or (s)kip this assignment?').strip().lower()
+                                if user_input in ['a', 'r', 's']:
+                                    break
+                            marked_reason = input('Reason for marking? (Can be left blank.)').strip()
+                            marked_assignments.append([assignment_id, user_input, marked_reason])
```

Comment:
For small, ad hoc data structures, I find dictionaries a little more future-proof: reordering or changing the elements of a list silently introduces a bug, but changing the keys in a dictionary throws a key error. Using a dictionary, this line might look like:

```python
marked_assignments.append({
    'assignment_id': assignment_id,
    'action': user_input,
    'reason': marked_reason
})
```

Reply:
Done